dll/cpl/sysdm/virtmem.c

```diff
@@ -151,44 +193,48 @@ ParseMemSettings(PVIRTMEM pVirtMem)
         {
             MinSize = -1;
             MaxSize = -1;
 
             /* Does drive match the one in the registry ? */
-            if (!_tcsncmp(pVirtMem->szPagingFiles, szDrive, 2))
+            if (_tcsnicmp(pVirtMem->szPagingFiles, szDrive, 2) == 0)
             {
                 GetPageFileSizes(pVirtMem->szPagingFiles,
                                  &MinSize,
                                  &MaxSize);
             }
 
-            pVirtMem->Pagefile[PgCnt].OldMinSize = MinSize;
-            pVirtMem->Pagefile[PgCnt].OldMaxSize = MaxSize;
-            pVirtMem->Pagefile[PgCnt].NewMinSize = MinSize;
-            pVirtMem->Pagefile[PgCnt].NewMaxSize = MaxSize;
-            pVirtMem->Pagefile[PgCnt].bUsed = TRUE;
-            lstrcpy(pVirtMem->Pagefile[PgCnt].szDrive, szDrive);
-
+            PageFile = &pVirtMem->PageFile[PgCnt];
+            PageFile->OldMinSize = MinSize;
+            PageFile->OldMaxSize = MaxSize;
+            PageFile->NewMinSize = MinSize;
+            PageFile->NewMaxSize = MaxSize;
+            PageFile->bIsNotFAT  = TRUE; /* Suppose this is not a FAT volume */
+            PageFile->bUsed = TRUE;
+            lstrcpy(PageFile->szDrive, szDrive);
 
             /* Get the volume label if there is one */
             if (GetVolumeInformation(DrivePtr,
-                                     szVolume,
-                                     MAX_PATH + 1,
-                                     NULL,
-                                     NULL,
-                                     NULL,
-                                     NULL,
-                                     0))
+                                     szVolume, _countof(szVolume),
+                                     NULL, NULL, NULL,
+                                     szFSName, _countof(szFSName)))
             {
-                pVirtMem->Pagefile[PgCnt].pszVolume = HeapAlloc(GetProcessHeap(),
-                                                                0,
-                                                                (_tcslen(szVolume) + 1) * sizeof(TCHAR));
-                if (pVirtMem->Pagefile[PgCnt].pszVolume != NULL)
-                    _tcscpy(pVirtMem->Pagefile[PgCnt].pszVolume, szVolume);
+                PageFile->pszVolume = HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY,
+                                                (_tcslen(szVolume) + 1) * sizeof(TCHAR));
+                if (PageFile->pszVolume != NULL)
+                    _tcscpy(PageFile->pszVolume, szVolume);
+
+                /*
+                 * Check whether the volume is FAT, which cannot support files
+                 * larger than 4GB (for FAT32 and FATX, and less for FAT16).
+                 * This will limit the maximum size of the pagefile this volume
+                 * can contain (see GetMaxPageFileSize()).
+                 */
+                PageFile->bIsNotFAT = (_tcsnicmp(szFSName, _T("FAT"), 3) != 0);
```

Comment:
I wonder if there's an API to ask filesystem for maximum file size :thinking:

Reply:
Sadly I've tried to find one but none showed up.